### ISSUE TITLE
Azure Table Check Always Unhealthy

### ISSUE THREAD

**Issue by junsaw** [Xabaril/AspNetCore.Diagnostics.HealthChecks#793]
Hey All,

I am using the Azure Table Check and its marked at unhealthy, same connection String is used for Blob and it works:

**Current version: 5.0.1** 

**Sample Code:** 
                services.AddHealthChecks()
                     .AddApplicationInsightsPublisher(Configuration["App:AppInsightsInstrumentationKey"])
                     .AddAzureBlobStorage(Configuration["Data:StorageAccountConnectionString"].ToString())
                     .AddAzureTable(Configuration["Data:StorageAccountConnectionString"].ToString(), Configuration["Data:AzureCountriesTableName"].ToString(), "Countries")
                     
**# TYPE healthcheck gauge**
healthcheck{healthcheck="Countries"} 0
healthcheck{healthcheck="azureblob"} 2

                     
Any help will be much appreciated.

Thanks Alot 
.\J

**Comment by unaizorrilla** [comment]
I try to review this a soon as posible!

**Comment by junsaw** [comment]
@unaizorrilla  - Let me know if there is any further information I can provide.

**Comment by unaizorrilla** [comment]
Hi @junsaw 

I just create a sample using latest package and all is working fine on my side! 

Here is the sample I used, obviously change your account and secret for your account https://gist.github.com/unaizorrilla/a1ca79dbd8dabb5923f3ea18425c0b81 

The package used on the sample 

<PackageReference Include="AspNetCore.HealthChecks.CosmosDb" Version="5.0.3" />
<PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="5.0.1" />

Please, can you create a repro that i can use, or check if my gist is working on your side?

**Comment by junsaw** [comment]
@unaizorrilla  let me try again, apologies for the delay, trying shortly.

**Comment by junsaw** [comment]
I validated again same error, I hard coded the value, as per your code. I will continue digging further, is there anything I can do to increase logging ?

**Comment by junsaw** [comment]
Some more information from Application Insights:



<html>
<body>


vent time | 7/9/2021, 10:17:06 PM (Local time) |  
-- | -- | --
Message | Method not found: 'System.String Azure.Data.Tables.Models.TableItem.get_TableName()'. |  
Exception type | System.MissingMethodException |  
Failed method | HealthChecks.CosmosDb.TableServiceHealthCheck+<CheckHealthAsync>d__7.MoveNext |  
Problem Id | System.MissingMethodException at HealthChecks.CosmosDb.TableServiceHealthCheck+<CheckHealthAsync>d__7.MoveNext |  
Assembly | HealthChecks.CosmosDb, Version=5.0.3.0, Culture=neutral, PublicKeyToken=null


</body>
</html>

**Comment by junsaw** [comment]
@unaizorrilla  anything further that can be done, I have tested this across several environments no success.  Any help will be much appreciated.

**Comment by junsaw** [comment]
Updated to latest version:  **5.0.4**
